src/uwtools/tests/fixtures/simple.sh

```diff
@@ -3,3 +3,5 @@ fruit=banana
 vegetable=tomato
 how_many=12
 dressing=balsamic
+# dressing=poison -- ensure that shlex ignores comments
+echo this will be ignored, too
```

Comment:
If `shlex` is not instructed to recognize comments, it will parse `#` as one token and `dressing=poison` as another token, and the latter will then be recognized as a key=value pair and added to the `SHConfig` object's dictionary. This comment is a tripwire to ensure that `shlex` comment recognition in the code isn't accidentally disabled.

Also note that `shlex` + the new `SHConfig` code is happy to ignore code that isn't key=value pairs. It's probably not good practice to parse real shell _scripts_ with `SHConfig`, but this demonstrates that at least some simple cases wouldn't be a problem.